save-frontend/src/main/kotlin/com/saveourtool/save/frontend/App.kt

```diff
@@ -31,109 +32,78 @@ import kotlinx.coroutines.await
- */
-@JsExport
-@OptIn(ExperimentalJsExport::class)
-class App : ComponentWithScope<PropsWithChildren, AppState>() {
-    init {
-        state.userInfo = null
-    }
+@Suppress(
+    "EMPTY_BLOCK_STRUCTURE_ERROR",
+    "TOO_LONG_FUNCTION",
+    "LongMethod",
+    "ComplexMethod",
+)
+val app = VFC {
+    val (userInfo, setUserInfo) = useState<UserInfo?>(null)
+    useRequest {
+        val userName: String? = get(
+            "${window.location.origin}/sec/user",
+            jsonHeaders,
+            loadingHandler = ::loadingHandler,
+            responseHandler = ::noopResponseHandler
+        ).run {
+            val responseText = text().await()
+            if (!ok || responseText == "null") null else responseText
+        }
 
-    override fun componentDidMount() {
-        getUser()
-    }
+        val globalRole: Role? = get(
+            "${window.location.origin}/api/$v1/users/global-role",
+            jsonHeaders,
+            loadingHandler = ::loadingHandler,
+            responseHandler = ::noopResponseHandler
+        ).run {
+            val responseText = text().await()
+            if (!ok || responseText == "null") null else Json.decodeFromString(responseText)
+        }
 
-    @Suppress("TOO_LONG_FUNCTION", "NULLABLE_PROPERTY_TYPE")
-    private fun getUser() {
-        scope.launch {
-            val userName: String? = get(
-                "${window.location.origin}/sec/user",
-                jsonHeaders,
-                loadingHandler = ::noopLoadingHandler,
-                responseHandler = ::noopResponseHandler
-            ).run {
-                val responseText = text().await()
-                if (!ok || responseText == "null") null else responseText
-            }
+        val user: UserInfo? = userName?.let { getUser(it) }
 
-            val globalRole: Role? = get(
-                "${window.location.origin}/api/$v1/users/global-role",
-                jsonHeaders,
-                loadingHandler = ::noopLoadingHandler,
-                responseHandler = ::noopResponseHandler
-            ).run {
-                val responseText = text().await()
-                if (!ok || responseText == "null") null else Json.decodeFromString(responseText)
-            }
+        val userInfoNew: UserInfo? = user?.copy(globalRole = globalRole) ?: userName?.let {
+            UserInfo(
+                name = userName,
+                globalRole = globalRole,
+            )
+        }
 
-            val user: UserInfo? = userName
-                ?.let { getUser(it) }
+        userInfoNew?.let {
+            setUserInfo(it)
+        }
+    }
 
-            val userInfoNew: UserInfo? = user?.copy(globalRole = globalRole) ?: userName?.let {
-                UserInfo(
-                    name = userName,
-                    globalRole = globalRole,
-                )
-            }
 
-            userInfoNew?.let {
-                setState {
-                    userInfo = userInfoNew
+    HashRouter {
```

Comment:
Will be changed in https://github.com/saveourtool/save-cloud/pull/2383

Reply:
yep, merged